icalevents/icalparser.py

```diff
@@ -431,7 +455,7 @@ def parse_rrule(component, tz=UTC):
 
             # Add exdates to the rruleset
             for exd in extract_exdates(component):
-                rule.exdate(exd)
+                rule.exdate(exd.replace(tzinfo=None) if type(rdtstart) is date else exd)
```

Comment:
Okay so we are setting tz to None if the DTSTART of the RRULE doesn't have a time zone?

Reply:
that is wrong - I changed the logic to use adjust_timezone and only remove tz if the component does not have one as well.